torch/nn/_functions/thnn/rnnFusedPointwise.py

```diff
@@ -10,37 +10,48 @@ def __init__(self):
     def forward(self, input_gate, hidden_gate, hx, ibias=None, hbias=None):
         if self.backend is None:
             self.backend = type2backend[type(input_gate)]
+
         hy = input_gate.new()
+        storage = input_gate.new().resize_(hx.numel() * 5)
+
+        self.hasBias = False
         if ibias is not None:
+            self.hasBias = True
             if ibias.dim() == 1:
-                ibias.unsqueeze_(0)
+                ibias = ibias.view(1, -1)
             if hbias.dim() == 1:
-                hbias.unsqueeze_(0)
+                hbias = hbias.view(1, -1)
 
         self.backend.GRUFused_updateOutput(
             self.backend.library_state,
-            input_gate, hidden_gate, ibias, hbias, hx, hy)
-        self.save_for_backward(input_gate, hidden_gate, ibias)
+            input_gate, hidden_gate, ibias, hbias, hx, hy, storage)
+
+        self.buffer = storage
+        self.igate_size = input_gate.size()
+        self.hgate_size = hidden_gate.size()
+
         return hy
 
     def backward(self, gradOutput):
         if self.backend is None:
             self.backend = type2backend[type(grad_output)]
-        gradInput = gradOutput.new()
-        input_gate, hidden_gate, bias = self.saved_tensors
 
-        igc = input_gate.clone()
-        hgc = hidden_gate.clone()
+        gradInputHx = gradOutput.new()
+        gradInInput = gradOutput.new().resize_(*self.igate_size)
+        gradInHidden = gradOutput.new().resize_(*self.hgate_size)
+
+        storage = self.buffer
```

Comment:
no need to assign it to a local, when it's only used once (just use an attr when calling the function)